docs/geps/gep-04.rst

```diff
@@ -0,0 +1,355 @@
+
+        Parameters
+        ----------
+        zu_verst_kapital_eink_tu
+            See :func:`zu_verst_kapital_eink_tu`.
+        abgelt_st_params
+            See params documentation :ref:`abgelt_st_params <abgelt_st_params>`.
+
+        Returns
+        -------
+
+        """
+        return abgelt_st_params["abgelt_st_satz"] * zu_verst_kapital_eink_tu
+
+The function :func:`abgelt_st_tu` requires the variable ``zu_verst_kapital_eink_tu``
+which is the amount of taxable capital income per tax unit (the latter is implied by the
+``_tu`` suffix, see :ref:`gep-1`). ``zu_verst_kapital_eink_tu`` must be provided by the
+user as a column of the input data or it has to be the name of another function.
+``abgelt_st_params`` is a dictionary of parameters related to the calculation of
+``abgelt_st_tu``.
+
+Another function, say
+
+.. code-block:: python
+
+    def soli_st_tu(
+        st_kind_freib_tu: FloatSeries,
+        anz_erwachsene_tu: IntSeries,
+        abgelt_st_tu: FloatSeries,
+        soli_st_params: dict,
+    ) -> FloatSeries:
+        ...
+
+may use ``abgelt_st_tu`` as an input argument. The DAG backend ensures that the function
+``abgelt_st_tu`` will be executed first.
+
+Note that the type annotations (e.g. `FloatSeries`) indicate the expected type of each
+input and the output of a function, see :ref:`gep-2`.
+
+
+Directed Acyclic Graph
+----------------------
+
+The relationship between functions and their input variables is a graph where nodes
+represent columns in the data. These columns must either be present in the data supplied
+to GETTSIM or they are computed by functions. Edges are pointing from input columns to
+variables, which require them to be computed.
+
+.. note::
+
+    GETTSIM allows to visualize the graph, see this `tutorial
+    <../visualize_the_system.ipynb>`_.
+
+The resulting structure is a special kind of graph, called a directed acyclic graph
+(DAG). It is directed because there are clearly inputs and outputs, i.e., there is a
+sense of direction. Acyclic means that there exist no path along the direction of the
+edges, where you start at some node and end up at the same node. Equivalently, a DAG has
+a topological ordering which is a sequence of nodes ordered from earlier to later in the
+sequence. The topological ordering is what defines the sequence in which the functions
+in the taxes and transfers system are evaluated. This ensures that the inputs are
+already computed before a function that requires them is called.
+
+In order to calculate a set of taxes and transfers, GETTSIM builds a DAG based on three
+inputs provided by the user:
+
+ - Input data.
+ - A set of functions representing the taxes and transfers system, which consist of the
+   ones pre-implemented in GETTSIM and potentially user-written additional functions.
+   Parameters of the taxes and transfers system (see :ref:`gep-3` will already be
+   partialled into these functions, so they can be ignored in the following). These
```

Comment:
The `partial` part is an implementation detail that I would not mention here.

Reply:
Agreed that it is an implementation detail. Seems to me that ignoring those arguments in the DAG is important for the text, though. New version in [576d03a](https://github.com/iza-institute-of-labor-economics/gettsim/pull/178/commits/576d03a80db263d18b25d3b4bd6272b4e7f0cf11) better?